src/azul/portal_service.py

```diff
@@ -0,0 +1,64 @@
+from copy import deepcopy
+from typing import (
+    Optional,
+    Sequence,
+    cast,
+)
+import json
+import logging
+
+import boto3
+from botocore.exceptions import ClientError
+
+from azul import config
+from azul.types import JSON
+
+log = logging.getLogger(__name__)
+
+
+class PortalService:
+
+    @staticmethod
+    def get_portal_integrations_db(default_db: Optional[Sequence[JSON]]) -> Sequence[JSON]:
+        """
+        Retrieve portal db from S3.
+        If no object is present at the location specified in azul.config and a default db is provided,
+        then it is uploaded and returned.
+        """
+
+        s3 = boto3.resource('s3')
+        bucket_name, object_key, charset = config.portal_integrations_db_params()
+        obj = s3.Object(bucket_name, object_key)
+        try:
+            retrieval = obj.get()
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchKey' and default_db is not None:
+                db = default_db
```

Comment:
```suggestion
                db = PortalService.stage_db(Plugin._unstaged_portal_integrations_db())
```

Reply:
I think there a couple reasons to avoid referencing `azul.project.hca.Plugin` in the `PortalService` class. 
First, it prevents `PortalService` from being used by any other classes. On a current practical note, this makes unit testing more tedious.

Reply:
I think I agree with @noah-aviel-dove here.